### DISCUSSION TITLE
How do i check if a container is running/stopped?

### DISCUSSION THREAD

**Discussion by darwin403** [testcontainers/testcontainers-node#541] [Q&A] [answered]
Sometimes is might be usefull to access to state of the container to check if it is running. 

For example, if a container has already been stopped and `container.stop()` is called again it would throw an error. The use case is when one more sub tests are closing containers and a global teradown needs to check if any contains are still running for a final all containers stop.

Any thoughts on this?

**Comment by cristianrgreco** [comment]
Hi @darwin403

It is a good question. 

The lifecycle of the container is controlled through its types, i.e GenericContainer -> StartedGenericContainer -> StoppedGenericContainer. From your example it seems you don't really use these types, or at least based on the way the test suites are structured, you don't care about them. 

Correct me if I'm wrong but what would help you is making `stop` idempotent, i.e if the container is already stopped, do nothing.

I am happy to implement this as I think generally it is prefereable. LMK if I've misunderstood something/if that works for you :slightly_smiling_face:

**Reply by cristianrgreco** [reply, marked as answer]
This has been implemented in https://github.com/testcontainers/testcontainers-node/pull/544 and released in v9.6.0.

**Reply by darwin403** [reply]
This is perfect! Thank you.
